components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -809,6 +809,31 @@ export function createWalletApi () {
               : ['UNKNOWN_ERROR']
         }
       ),
+      getCoingeckoId: query<
+        string,
+        Pick<BraveWallet.BlockchainToken, 'chainId' | 'contractAddress'>
+      >({
+        queryFn: async (
+          { chainId, contractAddress },
+          api,
+          extraOptions,
+          baseQuery
+        ) => {
+          const { blockchainRegistry } = baseQuery(undefined).data
+          const { coingeckoId } = await blockchainRegistry.getCoingeckoId(
+            chainId,
+            contractAddress
+          )
+
+          return {
+            data: coingeckoId
+          }
+        },
+        providesTags: (res, err, { chainId, contractAddress }) =>
+          err
+            ? ['CoingeckoId', 'UNKNOWN_ERROR']
+            : [{ type: 'CoingeckoId', id: `${chainId}-${contractAddress}` }]
+      }),
```

Comment:
Do we need providesTags sections for this api?
afaik it only makes sense if there are multiple apis dealing with same 'entity' but we have only one

Reply:
We will want a tag for this in the case that we want to invalidate a coingecko id for a token

https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#re-fetching-after-mutations-by-invalidating-cache-tags